docs/DeveloperGuide.md

```diff
@@ -1,9 +1,123 @@
 # Developer Guide
 
 ## Design & Implementation
-
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+![Architecture Diagram](https://github.com/DeetoMok/tp/raw/d14fb309c9f816a46ecd19ed26f74b8c8ac1ac03/Architecture.png)
+
+The Architecture Diagram given above explains the high-level design of the Module Manager Application.
+
+Module Manager consists of a main class called Duke responsible for
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other
+* At shut down: Shuts down the components and invokes cleanup method where necessary
+
+The other components involved are:
+
+`UI`: The user interface of the application
+
+`Storage`: Reads data from, and writes data to, the hard disk
+
+#### UI component
+The `UI` component consists of a `Ui` class that stores all user interaction output data. 
+It displays all user interactions to the user based on the command line inputs received.
+
+The `UI` component,
+
+*   Executes user commands using the `Logic` component
+
+#### Logic component
+The `Logic` component 
+
+1. `Logic` uses the `Parser` class to parse the user command.
+2. `Parser` then returns a specific command class e.g. `AddCommand`, `FindCommand` etc. 
+which is executed by the main class `Duke`.
+All these command classes inherits from the abstract `Command` class.
+3. The command execution can affect the Model (e.g. adding a module in ModuleList)
+4. The result of the command execution is passed back to the Ui.
+5. In addition, the command execution can also instruct the `Ui` to perform certain actions, 
+such as displaying help to the user.
+
+#### Model component
+The `Model` component
+* Stores a `Person` object that represents the user's details. (e.g. `totalCap`, `totalModuleCreditCompleted`)
+* Stores the `ModuleList` and `SemesterList` Data
+* Does not depend on any of the other three components
```

Comment:
I was wondering if Model component is not related to other three components, then how would a Person object, which is stored in Model Component, have their CAP calculated? Maybe "does not depend" is not the correct phrase here?